### PR TITLE
Add Integer property literals.

### PR DESCRIPTION
**Pull request by mikeurbach**
This defines literal property type expressions in general, and Integer property type literals specifically. Examples are added to show example Integer literals, as well as using literals in propassign statements.

### CODE REVIEW

**Review by seldridge — APPROVED** [review]
I'm inclined to say that we should allow this for any of the radix-specified integer literals, too. E.g., `Integer(0hdeadbeef)`. However, we can add this later. WDYT?

File: spec.md [hunk 1]
```diff
@@ -2485,6 +2495,23 @@ SInt("h-2A")
...
+
+### Integer Property Literal Expresssions
+
+A literal `Integer` property type expression can be created from an integer
+literal. The following examples show literal `Integer` property type
+expressions.
+
+``` firrtl
+Integer(42)
+Integer(-42)
```
Comment (mwachs5) [inline spec.md:2512]: can you write in hex or any other formats?

**Comment by seldridge** [comment]
Semi-related, I pushed a spec patch change underneath this. The examples just above this section in the spec were still using string-encoded integer literals. I'm not sure how that slipped through. I thought I fully removed those in 3.0.0. 🤦

**Comment by mikeurbach** [comment]
We could support radix-specified integer literals. I know there was a reason to support both integer literals and radix-specified literals for UInt and SInt, but I don't think there is any need to support multiple ways of doing it for properties. So I guess I'd prefer to just start with one format. Would radix-specified be better? I picked base-10 integer literals because the initial Chisel and CIRCT support were using base-10 integers, but we can change it.

**Comment by seldridge** [comment]
Keep it with normal base-10, non-radix specified like you have it, i.e., `42` and `-42`. 👍

**Pull request merged**